packages/transaction-controller/src/TransactionController.ts

```diff
@@ -2747,4 +3034,216 @@ export class TransactionController extends BaseControllerV1<
       log('Error while updating post transaction balance', error);
     }
   }
+
+  #getEthQuery({
+    networkClientId,
+    chainId,
+  }: {
+    networkClientId?: NetworkClientId;
+    chainId?: Hex;
+  }): EthQuery {
+    // if multichain is disabled, use the global ethQuery
+    if (!this.enableMultichain) {
+      return this.ethQuery;
+    }
+    let networkClient: NetworkClient | undefined;
+
+    if (networkClientId) {
+      try {
+        networkClient = this.getNetworkClientById(networkClientId);
+      } catch (err) {
+        log('failed to get network client by networkClientId');
+      }
+    }
+
+    if (!networkClient && chainId) {
+      try {
+        networkClientId = this.findNetworkClientIdByChainId(chainId);
+        networkClient = this.getNetworkClientById(networkClientId);
+      } catch (err) {
+        log('failed to get network client by chainId');
+      }
+    }
```

Comment:
if the passed in networkClientId doesn't resolve to a network client (because it was removed and no longer exists), we want to use the chainId as a fallback to resolving a network client